Make ServiceBrowser updates threadsafe on pypy

```
self = <ServiceBrowser(zeroconf-ServiceBrowser-_my-service._tcp-123145393102848, stopped daemon 123145393102848)>

    def update_records_complete(self) -> None:
        """Called when a record update has completed for all handlers.
    
        At this point the cache will have the new records.
        """
>       self._handlers_to_call.update(self._pending_handlers)
E       RuntimeError: dictionary changed size during iteration

zeroconf/_services/__init__.py:372: RuntimeError
```

https://github.com/jstasiak/python-zeroconf/pull/706/checks?check_run_id=2852655806

--- a/zeroconf/_services/__init__.py
+++ b/zeroconf/_services/__init__.py
@@ -369,8 +369,15 @@ class _ServiceBrowserBase(RecordUpdateListener):
 
         At this point the cache will have the new records.
         """
-        self._handlers_to_call.update(self._pending_handlers)
-        self._pending_handlers.clear()
+        # Cannot use .update here since PyPy can fail with
+        # RuntimeError: dictionary changed size during iteration
+        # for threaded ServiceBrowsers
+        while self._pending_handlers:
+            try:
+                (name_type, state_change) = self._pending_handlers.popitem(False)
+            except KeyError:
+                return
+            self._handlers_to_call[name_type] = state_change
 
     def cancel(self) -> None:
         """Cancel the browser."""

